Sources/ArcGISToolkit/Components/BasemapGallery/BasemapGallery.swift

```diff
@@ -117,43 +102,51 @@ public struct BasemapGallery: View {
                 } message: { item in
                     Text(item.message)
                 }
-                .frame(height: min(contentHeight, geometry.size.height))
-                .esriBorder()
+                .frame(
+                    width: geometry.size.width,
+                    height: geometry.size.height
+                )
         }
-        .frame(width: galleryWidth)
     }
 }
 
 private extension BasemapGallery {
     /// Creates a gallery view.
+    /// - Parameter containerWidth: The width of the container holding the gallery.
     /// - Returns: A view representing the basemap gallery.
-    func makeGalleryView() -> some View {
+    func makeGalleryView(_ containerWidth: CGFloat) -> some View {
         ScrollView {
             switch style {
-            case .automatic:
+            case .automatic(let maxGridItemWidth):
                 if isRegularWidth {
-                    makeGridView()
+                    makeGridView(containerWidth, maxGridItemWidth)
                 } else {
                     makeListView()
                 }
-            case .grid:
-                makeGridView()
+            case .grid(let maxItemWidth):
+                makeGridView(containerWidth, maxItemWidth)
             case .list:
                 makeListView()
             }
         }
     }
     
     /// The gallery view, displayed as a grid.
+    /// - Parameters:
+    ///   - containerWidth: The width of the container holding the grid view.
+    ///   - maxItemWidth: The maximum allowable width for an item in the grid. Defaults to `300`.
     /// - Returns: A view representing the basemap gallery grid.
-    func makeGridView() -> some View {
+    func makeGridView(_ containerWidth: CGFloat, _ maxItemWidth: CGFloat) -> some View {
         internalMakeGalleryView(
             columns: Array(
                 repeating: GridItem(
                     .flexible(),
                     alignment: .top
                 ),
-                count: 3
+                count: max(
+                    1,
```

Comment:
I would make this `2`, so it would always be a grid. Otherwise the `grid` style could look like a list.
```suggestion
                    2,
```

Reply:
I'd be fine with 2 if you prefer, my only concern is that things can get a bit cramped if automatic is used and the component is provided little width.

The following code produces the results below:
```swift
.overlay(alignment: .trailing) {
    if showBasemapGallery {
        BasemapGallery(items: basemaps, geoModel: map)
            .style(.automatic(maxGridItemWidth: 200))
            .esriBorder()
            .padding()
            .frame(width: width)
    }
}
```

`max(2,...)` with a total width of 140 provided:
![max2](https://user-images.githubusercontent.com/16397058/230485839-d37603b6-f940-4fe0-b8b7-4c670c9e9d82.png)

`max(1,...)` with a total with of 140 provided:
![max1](https://user-images.githubusercontent.com/16397058/230485991-b69f02f5-a530-4a60-810e-ae020939e94e.png)

Reply:
I'm Ok with 1.